### PR TITLE
refactor: use images from AWS Private ECR instead of public registries

### PR DESCRIPTION
**Pull request by prasunna09**

### CODE REVIEW

File: dependencies/code_builder/buildspec.yml [hunk 1]
```diff
@@ -8,38 +8,61 @@ phases:
...
+          local image_tag="$2"
+          aws ecr batch-delete-image --repository-name "$repo_name" --image-ids imageTag="$image_tag"
+        }
 
-        for IMAGE in "${PULL_FROM_K8S[@]}"; do
-          aws ecr create-repository --repository-name $IMAGE --image-scanning-configuration scanOnPush=true
+        for IMAGE in "${PULL_FROM_DOCKER[@]}"; do
+          REPO_NAME="$IMAGE"
+          if ! repository_exists "$REPO_NAME"; then
+              aws ecr create-repository --repository-name "$REPO_NAME" --image-scanning-configuration scanOnPush=true
```
Comment (jagan-jaya) [inline dependencies/code_builder/buildspec.yml:31]: will it ignore if the repo already present?

File: install.sh [hunk 2]
```diff
@@ -423,6 +423,7 @@ if [[ "$INSTALLATION_MODE" == 2 ]]; then
         aws iam delete-role --role-name $ROLE_NAME 2>/dev/null
         cdk bootstrap aws://$AWS_ACCOUNT_ID/$AWS_DEFAULT_REGION -c aws_arn=$AWS_ARN
     fi
+    echo `cdk deploy --require-approval never -c db_pass=$DB_PASS -c admin_api_key=$ADMIN_API_KEY -c aws_arn=$AWS_ARN -c master_enc_key=$MASTER_ENC_KEY -c vpn_ips=$VPN_IPS -c base_ami=$base_ami -c envoy_ami=$envoy_ami -c squid_ami=$squid_ami $LOCKER;`
```
Comment (jagan-jaya) [inline install.sh:426]: this script might get executed, is it needed?

File: lib/aws/eks.ts [hunk 3]
```diff
@@ -451,24 +451,23 @@ export class EksStack {
...
-              image: `juspaydotin/hyperswitch-producer:v1.107.0-standalone`
+              image: `${privateEcrRepository}/juspaydotin/hyperswitch-producer:v1.107.0-standalone`
             },
             consumer: {
-              image: `juspaydotin/hyperswitch-consumer:v1.107.0-standalone`
+              image: `${privateEcrRepository}/juspaydotin/hyperswitch-consumer:v1.107.0-standalone`
             },
             controlCenter: {
-              image: `juspaydotin/hyperswitch-control-center:v1.17.0`
+              image: `${privateEcrRepository}/juspaydotin/hyperswitch-control-center:v1.29.9`
```
Comment (jagan-jaya) [inline lib/aws/eks.ts:470]: is this version tested in local?

File: lib/aws/eks.ts [hunk 4]
```diff
@@ -496,8 +495,8 @@ export class EksStack {
                 kms_jwekey_locker_encryption_key2: kmsSecrets.kms_jwekey_locker_encryption_key2,
                 kms_jwekey_locker_decryption_key1: kmsSecrets.kms_jwekey_locker_decryption_key1,
                 kms_jwekey_locker_decryption_key2: kmsSecrets.kms_jwekey_locker_decryption_key2,
-                kms_jwekey_vault_encryption_key: kmsSecrets.kms_jwekey_vault_encryption_key,
-                kms_jwekey_vault_private_key: kmsSecrets.kms_jwekey_vault_private_key,
+                kms_jwekey_vault_encryption_key: locker?.locker_ec2.locker_pair.public_key || kmsSecrets.kms_jwekey_vault_encryption_key,
```
Comment (jagan-jaya) [inline lib/aws/eks.ts:498]: undo this change

File: lib/aws/eks.ts [hunk 5]
```diff
@@ -496,8 +495,8 @@ export class EksStack {
...
+                kms_jwekey_vault_private_key: locker?.locker_ec2.tenant.private_key || kmsSecrets.kms_jwekey_vault_private_key,
```
Comment (jagan-jaya) [inline lib/aws/eks.ts:499]: undo this change

File: values.yaml
Comment (jagan-jaya) [inline values.yaml:1]: this file not supposed to be in CDK. please remove it immediately from main repo if present

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: dependencies/code_builder/buildspec.yml [hunk 6]
```diff
@@ -8,38 +8,61 @@ phases:
...
+          fi
+          if image_exists "$REPO_NAME" "latest"; then
+              delete_image "$REPO_NAME" "latest"
+          fi
+          docker pull "$IMAGE" && docker tag "$IMAGE" "$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$IMAGE:latest" && docker push "$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$IMAGE:latest"
         done
```
Comment (nitesh-balla) [inline dependencies/code_builder/buildspec.yml:37]: Why are we doing this?

**Review by jagan-jaya — APPROVED** [review]

File: lib/aws/userdata.sh [hunk 7]
```diff
@@ -26,7 +26,7 @@ ROUTER__SERVER__BASE_URL=$(curl ifconfig.me)
 ROUTER__SECRETS__ADMIN_API_KEY={{admin_api_key}}
 EOF
 
-docker run -d --env-file .env -p 80:8080 -v `pwd`/:/local/config juspaydotin/hyperswitch-router:v1.105.0-standalone ./router -f /local/config/production.toml
+docker run -d --env-file .env -p 80:8080 -v `pwd`/:/local/config juspaydotin/hyperswitch-router:v1.107.0-standalone ./router -f /local/config/production.toml
```
Comment (nitesh-balla) [inline lib/aws/userdata.sh:30]: Please revert these changes back for now.

File: dependencies/code_builder/buildspec.yml [hunk 8]
```diff
@@ -8,38 +8,61 @@ phases:
...
+          local repo_name="$1"
+          local image_tag="$2"
+          aws ecr describe-images --repository-name "$repo_name" --image-ids imageTag="$image_tag" >/dev/null 2>&1
+        }
+
+        delete_image() {
+          local repo_name="$1"
+          local image_tag="$2"
+          aws ecr batch-delete-image --repository-name "$repo_name" --image-ids imageTag="$image_tag"
+        }
```
Comment (nitesh-balla) [inline dependencies/code_builder/buildspec.yml:26]: We have tag immutability disabled for the repositories that are created, So, we can upload the same tag multiple times into the repository without any error! Deletion isn't required.

**Review by nitesh-balla — CHANGES_REQUESTED** [review]

**Review by nitesh-balla — APPROVED** [review]

**Review by nitesh-balla — APPROVED** [review]

**Pull request merged**